phoenix-core/src/main/java/org/apache/phoenix/hbase/index/Indexer.java

```diff
@@ -579,16 +798,79 @@ public void postBatchMutateIndispensably(ObserverContext<RegionCoprocessorEnviro
 
   private void doPost(ObserverContext<RegionCoprocessorEnvironment> c, BatchMutateContext context) throws IOException {
       try {
-        doPostWithExceptions(c,context);
-        return;
+          doPostWithExceptions(c,context);
+          return;
       } catch (Throwable e) {
-        rethrowIndexingException(e);
+          rethrowIndexingException(e);
       }
       throw new RuntimeException(
-          "Somehow didn't complete the index update, but didn't return succesfully either!");
-    }
+              "Somehow didn't complete the index update, but didn't return succesfully either!");
+  }
 
   private void doPostWithExceptions(ObserverContext<RegionCoprocessorEnvironment> c, BatchMutateContext context)
+            throws IOException {
```

Comment:
This function looks like doPreWithExceptions. Can we refactor and use the same function?

Reply:
I will do that

Reply:
Done